knative-operator/pkg/controller/knativeserving/knativeserving_controller.go

```diff
@@ -130,6 +140,31 @@ func (r *ReconcileKnativeServing) configure(instance *servingv1alpha1.KnativeSer
 	return nil
 }
 
+// deleteVirtualService removes obsoleted VirtualServices.
+func (r *ReconcileKnativeServing) deleteVirtualService(instance *servingv1alpha1.KnativeServing) error {
+	labelSelector := labels.NewSelector()
+	req, err := labels.NewRequirement(routeLabelKey, selection.Exists, nil)
+	if err != nil {
+		return fmt.Errorf("failed to create requirement for label: %w", err)
+	}
+	listOpts := &client.ListOptions{LabelSelector: labelSelector.Add(*req)}
+	list := &v1alpha3.VirtualServiceList{}
+	ctx := context.TODO()
+	if err := r.client.List(ctx, listOpts, list); err != nil {
+		return err
```

Comment:
I think we need to ignore an `apierrs.NoMatchError` in case no VirtualService CRD is installed at all.

Reply:
While you're at it, we can add `failed to list VirtualServices` maybe?

Reply:
Yes, we must ignore the NoMatchError. added the error handling and output the info message below:

```
{"level":"info","ts":1584710238.468574,"logger":"knative.openshift.controller","msg":"failed to list VirtualServices"}
```